flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -544,6 +544,12 @@ private CatalogBaseTable instantiateCatalogTable(Table hiveTable, HiveConf hiveC
 					properties,
 					comment);
 		} else {
+			List<String> primaryKey = client.getPrimaryKey(hiveTable.getDbName(), hiveTable.getTableName(),
+					HiveTableUtil.relyConstraint((byte) 0));
+			// Need a better place to hold the PK constraints. Put it as a property for now.
```

Comment:
there is already a pr: [[FLINK-12846][table-common] Carry primary key information in TableSchema ](https://github.com/apache/flink/pull/10213) to solve this, add this info to the comments ?